libkbfs/block_retrieval_worker.go

```diff
@@ -67,8 +67,11 @@ func (brw *blockRetrievalWorker) HandleRequest() (err error) {
 	}
 
 	// Create a new block of the same type as the first request
-	typ := reflect.TypeOf(retrieval.requests[0].block).Elem()
-	block := reflect.New(typ).Interface().(Block)
+	var block Block
+	if retrieval.requests[0].block != nil {
```

Comment:
Do some tests pass in a `nil` block or something?  This seems like an odd check to need.  Seems like the block retriever wouldn't be able to handle a `nil` block anyway.